doc/api/deprecations.md

```diff
@@ -0,0 +1,491 @@
+
+The `CryptoStream.prototype.readyState` property is deprecated and should not
+be used.
+
+<a id="DEP00005"></a>
+### DEP00005: Buffer() constructor
+
+Type: Documentation-only
+
+The `Buffer()` and `new Buffer()` constructors are considered to be deprecated
+due to API usability issues that can potentially lead to accidental security
+issues.
+
+As an alternative, use of the following methods of constructing `Buffer` objects
+is strongly recommended:
+
+* `Buffer.alloc(size[, fill[, encoding]])` - Create a `Buffer` with
+  *initialized* memory.
+* `Buffer.allocUnsafe(size)` - Create a `Buffer` with *uninitialized* memory.
+* `Buffer.allocUnsafeSlow(size)` - Create a `Buffer` with *uninitialized*
+   memory.
+* `Buffer.from(array)` - Create a `Buffer` with a copy of `array`
+* `Buffer.from(arrayBuffer[, byteOffset[, length]])` - Create a `Buffer` that
+   wraps the given `arrayBuffer`.
+* `Buffer.from(buffer)` - Create a `Buffer` that copies `buffer`.
+* `Buffer.from(string[, encoding])` - Create a `Buffer` that copies `string`.
+
+<a id="DEP00006"></a>
+### DEP00006: child\_process options.customFds
+
+Type: Runtime
+
+Within the `child_process` module's `spawn()`, `fork()`, and `exec()` methods,
+The `options.customFds` option is deprecated. The `options.stdio` option
+should be used instead.
+
+<a id="DEP00007"></a>
+### DEP00007: cluster worker.suicide
+
+Type: Runtime
+
+When using the `cluster` module, the `worker.suicide` property has been
+deprecated. Please use `worker.exitedAfterDisconnect` instead.
+
+<a id="DEP00008"></a>
+### DEP00008: require('constants')
+
+Type: Runtime
+
+The `constants` module has been deprecated. When requiring access to constants
+relevant to specific Node.js built in modules, developers should instead refer
+to the `constants` property exposed by the relevant module. For instance,
+`require('fs').constants` and `require('os').constants`.
+
+<a id="DEP00009"></a>
+### DEP00009: crypto.pbkdf2 without digest
+
+Type: Runtime
+
+Use of the `crypto.pbkdf2()` API without specifying a digest is deprecated.
+Please specify a digest.
+
+<a id="DEP00010"></a>
+### DEP00010: crypto.createCredentials
+
+Type: Runtime
+
+The `crypto.createCredentials()` API is deprecated. Please Use
+`tls.createSecureContext()` instead.
+
+<a id="DEP00011"></a>
+### DEP00011: crypto.Credentials
+
+Type: Runtime
+
+The `crypto.Credentials` class is deprecated. Please Uuse `tls.SecureContext`
+instead.
+
+<a id="DEP00012"></a>
+### DEP00012: Domain.dispose
+
+Type: Runtime
+
+`Domain.dispose()` is deprecated. Recover from failed I/O actions explicitly via
+error event handlers set on the domain instead.
+
+<a id="DEP00013"></a>
+### DEP00013: fs async function without callback
+
+Type: Runtime
+
+Calling an asynchronous function without callback is deprecated.
+
+<a id="DEP00014"></a>
+### DEP00014: fs.read legacy String interface
+
+Type: Runtime
+
+The `fs.read()` legacy String interface is deprecated. Use the Buffer API as
```

Comment:
We explicitly ask the readers to refer documentation, it would be better if we had the link here.